VSRAD.Package/Server/ActionRunner.cs

```diff
@@ -15,18 +16,29 @@
 
 namespace VSRAD.Package.Server
 {
+    public interface IActionRunController
+    {
+        CancellationToken CancellationToken { get; }
+
+        Task<bool> ShouldTerminateProcessOnTimeoutAsync(IList<ProcessTreeItem> processTree);
+
+        Task OpenFileInVsEditorAsync(string path, string lineMarker);
+    }
+
     public sealed class ActionRunner
     {
         private readonly ICommunicationChannel _channel;
         private readonly SVsServiceProvider _serviceProvider;
+        private readonly IActionRunController _controller;
         private readonly Dictionary<string, DateTime> _initialTimestamps = new Dictionary<string, DateTime>();
         private readonly ReadOnlyCollection<string> _debugWatches;
         private readonly IProject _project;
 
-        public ActionRunner(ICommunicationChannel channel, SVsServiceProvider serviceProvider, ReadOnlyCollection<string> debugWatches, IProject project)
+        public ActionRunner(ICommunicationChannel channel, SVsServiceProvider serviceProvider, IActionRunController controller, ReadOnlyCollection<string> debugWatches, IProject project)
```

Comment:
`SVsServiceProvider serviceProvider` should be removed from the constructor parameters; `_serviceProvider` should be removed as well. This will probably require updating `ActionRunnerTests`.